pkg/virt-operator/install-strategy/delete.go

```diff
@@ -278,24 +278,33 @@ func DeleteAll(kv *v1.KubeVirt,
 		if !ok {
 			return fmt.Errorf("couldn't cast object to SecurityContextConstraints: %+v", privSCCObj)
 		}
+
+		oldUsers := privSCC.Users
 		privSCCCopy := privSCC.DeepCopy()
+		users := privSCCCopy.Users
 
 		modified := false
-		users := privSCCCopy.Users
+
 		for _, acc := range sccPriv.ServiceAccounts {
 			removed := false
 			users, removed = remove(users, acc)
 			modified = modified || removed
 		}
 
 		if modified {
+			oldUserBytes, err := json.Marshal(oldUsers)
+			if err != nil {
+				return err
+			}
 			userBytes, err := json.Marshal(users)
 			if err != nil {
 				return err
 			}
 
-			data := []byte(fmt.Sprintf(`{"users": %s}`, userBytes))
-			_, err = scc.SecurityContextConstraints().Patch(sccPriv.TargetSCC, types.StrategicMergePatchType, data)
+			test := fmt.Sprintf(`{ "op": "test", "path": "/users", "value": %s }`, string(oldUserBytes))
```

Comment:
Just curious what exactly `test` operator is doing?

Reply:
@cynepco3hahue JSON patches have the ability to issue a `test` command to ensure the contents of the resource match an expected state before continuing. this allows the abilty to do something like "test the users = [bob]" and the "replace the contents of users with [bob, alice]" ... and since the json patch is applied all at once this all happens in one step. 

if you just did the replace command without checking the contents you could accidentally clobber a conflicting change.

Reply:
We don't own the SCC object, but our PATCH is attempting to replace the entire "users" array.

The "test" ensures that we aren't colliding with another process that's also attempting to modify "users".  Test means that if "users" isn't exactly equal to what we think it shoudl be equal to before the "replace" that the Patch operation should fail.

so, if another process sneaks in and adds a user to the users array right before our Patch operation, the "test" op would fail causing the Patch to fail. 

Basically "test" is a safety mechanism.

Reply:
Thanks for detailed answer